src/tests.c

```diff
@@ -182,8 +182,10 @@ void run_context_tests(int use_prealloc) {
     ecount2 = 10;
     secp256k1_context_set_illegal_callback(vrfy, counting_illegal_callback_fn, &ecount);
     secp256k1_context_set_illegal_callback(sign, counting_illegal_callback_fn, &ecount2);
-    secp256k1_context_set_error_callback(sign, counting_illegal_callback_fn, NULL);
-    CHECK(vrfy->error_callback.fn != sign->error_callback.fn);
+    /* set error callback (to a function that still aborts in case malloc() fails in secp256k1_context_clone() below) */
+    secp256k1_context_set_error_callback(sign, secp256k1_default_illegal_callback_fn, NULL);
```

Comment:
I see now that in other places we do `secp256k1_context_set_error_callback(sign, NULL, NULL);` instead of explicitly specifying the function

Reply:
NULL sets it to the default.

Reply:
I missed the fact that this isn't actually the default function. but both will abort, so I think that was by mistake?

Reply:
No, this is intentional. The purpose of the next line is to verify that the setter actually sets (and isn't a no-op). So it needs to be set to something different to perform that test.

Some additional tests added after the initial setter-test were introduced call the clone function so they actually need the context to be functional.  They still logically belong using this context, however, because they're calling clone to verify (among other things) that the clone copies the configured setter function.

Your comments suggest that this is unclear enough that it deserves an explicit comment.  /* This uses secp256k1_default_illegal_callback_fn instead of secp256k1_default_error_callback_fn because it needs to invoke it with non-default function as the argument for the following setter and clone tests, but the configured function should still work correctly should malloc happen to get called during the tests. */

Reply:
3 lines above we setted the error callback to be `counting_illegal_callback_fn` so I do think that setting back to the default still tests the same thing.
But it doesn't really matter, so you can ignore my comment

Reply:
I see now that I keep getting confused between error and illegal :) your response makes sense. thanks.

Reply:
No three lines above it sets the illegal cb, not the error cb.

Reply:
so you stick to your ACK?

Reply:
yep. it was a nit anyway, and it turned out as a wrong nit :)
(but feel free to add a comment similar to what @gmaxwell was suggesting)